### ISSUE TITLE
email registration always as 'guest'`

### ISSUE THREAD

**Issue by kkplein** [inverse-inc/packetfence#969]
On packetfence 5.3.1, inline, using email registration I would like different roles to be applied, based on the kind of email address the user used. But all registrations end up as type 'guest', and the roles I configured are never applied:

> Oct 19 15:20:25 httpd.portal(14054) INFO: Email sent to test@heupink.com (our.domain.com: Email activation required) (pf::activation::**ANON**)
> Oct 19 15:23:40 httpd.portal(14041) INFO: [60:67:20:5d:74:98] Activation code sent to email test@heupink.com from test@heupink.com successfully verified.  for activation type: guest (pf::activation::validate_code)
> And not the expected role "heupink_com" It seems always the last 'catch all' rule is applied.

This is my authentication.conf:

> root@pf:/usr/local/pf/conf# cat authentication.conf
> [local]
> description=Local Users
> type=SQL
> 
> [file1]
> description=Legacy Source
> stripped_user_name=yes
> path=/usr/local/pf/conf/admin.conf
> type=Htpasswd
> 
> [file1 rule admins]
> description=All admins
> match=all
> action0=set_access_level=ALL
> 
> [email]
> description=Email-based registration
> email_activation_timeout=10m
> type=Email
> create_local_account=no
> allow_localdomain=yes
> 
> [email rule unu]
> description=mpp students, for example
> match=all
> action0=set_role=unu_edu
> action1=set_access_duration=365D
> condition0=user_email,ends,unu.edu
> 
> [email rule heupink]
> description=to test access duration
> match=all
> action0=set_role=heupink_com
> action1=set_access_duration=12h
> condition0=user_email,contains,@heupink.com
> 
> [email rule maastrichtuniversity]
> description=unimaas accounts
> match=any
> action0=set_role=unimaas
> action1=set_access_duration=365D
> condition0=user_email,ends,maastrichtuniversity.nl
> condition1=user_email,ends,unimaas.nl
> 
> [email rule catchall]
> description=others one month
> match=all
> action0=set_role=guest
> action1=set_access_duration=30D
> 
> [null]
> description=Null Source
> type=Null
> email_required=no
> Note: I also tried a condition like: condition0=user_email,ends,heupink.com, and it also does not apply.

I also tried bin/pfcmd configreload but it made no difference.

**Comment by fdurand** [comment]
Did you tried:

```
[email rule heupink]
description=to test access duration
match=all
action0=set_role=heupink_com
action1=set_access_duration=12h
condition0=user_email,contains,\@heupink.com
```

**Comment by kkplein** [comment]
Nope, but I have tried
condition0=user_email,contains,heupink.com
(realising that this would match too much) and it did not seem to work...
I'm in no position to try your suggestion right now, because I am not onsite, and afaik there is no way to simulate this from the command line.. (pftest does not work to test this, i guess)

**Comment by Latruncul** [comment]
I've the same issue on 5.4

@fdurand 
Did you tried:

```
[email rule heupink]
description=to test access duration
match=all
action0=set_role=heupink_com
action1=set_access_duration=12h
condition0=user_email,contains,\@heupink.com
```

Tried, but this didn't help

**Comment by dwlfrth** [comment]
Due to the fact that 'user_email' is an attribute of the EmailSource, condition matching must be done in subclass, which we don't.

**Comment by Latruncul** [comment]
Any news?

**Comment by heupink** [comment]
Is this something that anyone is looking at..? Any ideas when (and if) this will be fixed?

**Comment by louismunro** [comment]
Yes, someone is looking into it.
Don't hold your breath though.
It will not be part of 5.5. 
_Maybe_ 5.6. 

The issue is that the authentication source is not designed to do that. 
We should not allow creating rules like those on it.

It looks like a valid feature request though. 
We will definitely look into implementing it but are not ready to give an ETA at this point.

**Comment by kkplein** [comment]
Would it speed it up if we spend a support incident on it? (perhaps together with Latruncul?)
So the above also means that you intentions are to actually REMOVE the functionality from the gui in 5.5?

**Comment by louismunro** [comment]
This can't make it into 5.5 regardless.
It is too late in that release cycle to change that.
It could be in 5.6 though.

My comments with regards to the GUI only meant that it is currently misleading, as there is no point allowing the user to create a rule that will never match. 

Short answer is yes, a ticket would help. I can't yet comment on precisely how much time and work this would require anyway so that would have to be discussed, but it would certainly focus things on this one.

**Comment by jrouzierinverse** [comment]
Fixed in 27a79aec77daa4a628ab2517a77c6c967b60908f and 0b38f1a373841fd3d8ab2aa7c88e07c5c5f31799
